Graphs/BinaryLifting.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Author: Adrito Mukherjee
+ * Binary Lifting implementation in Javascript
+ * Binary Lifting is a technique that is used to find the kth ancestor of a node in a rooted tree with N nodes
+ * The technique requires preprocessing the tree in O(N log N) using dynamic programming
+ * The techniqe can answer Q queries about kth ancestor of any node in O(Q log N)
+ * It is faster than the naive algorithm that answers Q queries with complexity O(Q K)
+ * It can be used to find Lowest Common Ancestor of two nodes in O(log N)
+ * Tutorial on Binary Lifting: https://codeforces.com/blog/entry/100826
+ */
+
+class BinaryLifting {
+  constructor (root, tree) {
+    this.connections = {}
+    this.up = {} // up[node][i] stores the 2^i-th  parent of node
```

Comment:
Using a dictionary - which stringifies all its keys - here is dirty (as nodes won't necessarily be strings; they'll be objects). Please use a `Map` instead.

Reply:
done